### PR TITLE
[SPARK-15560] [Mesos] Queued/Supervise drivers waiting for retry drivers disappear for kill command in Mesos mode

### PR DESCRIPTION
**Pull request by devaraj-kavali**
## What changes were proposed in this pull request?

With the patch, it moves the drivers from Queued Drivers/Supervise drivers waiting for retry section to Finished Drivers section when they get killed.
## How was this patch tested?

I have verified it manually by checking the Mesos Dispatcher UI while simulating this scenario.

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosClusterScheduler.scala [hunk 1]
```diff
@@ -188,10 +188,10 @@ private[spark] class MesosClusterScheduler(
         mesosDriver.killTask(task.taskId)
         k.success = true
         k.message = "Killing running driver"
-      } else if (removeFromQueuedDrivers(submissionId)) {
```
Comment (tnachen) [inline core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosClusterScheduler.scala]: We should just rename and change the existing function, I Don't think it's being used elsewhere?

Comment (devaraj-kavali) [inline core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosClusterScheduler.scala]: Thanks @tnachen for looking into this, I see it is being used in other places.

```
      queuedDrivers
        .filter(d => launchedDrivers.contains(d.submissionId))
        .foreach(d => removeFromQueuedDrivers(d.submissionId))

```

```
      // Then we walk through the queued drivers and try to schedule them.
      scheduleTasks(
        copyBuffer(queuedDrivers),
        removeFromQueuedDrivers,
        currentOffers,
        tasks)
```

**Comment by tnachen** [comment]
ok to test

**Comment by devaraj-kavali** [comment]
https://amplab.cs.berkeley.edu/jenkins/job/SparkPullRequestBuilder/59989/testReport/

`org.apache.spark.scheduler.BlacklistIntegrationSuite.Bad node with multiple executors, job will still succeed with the right confs`

This test is passing in my local environment and also doesn't seem to be related this change.

@tnachen can we retest it?

**Comment by tnachen** [comment]
jenkins please retest

**Comment by devaraj-kavali** [comment]
@tnachen Can you check this?

**Comment by devaraj-kavali** [comment]
@mgummelt /@tnachen, can you have a look into this?

**Comment by mgummelt** [comment]
A killed driver never finished, so it shouldn't be added to the finished set.

**Comment by devaraj-kavali** [comment]
Thanks @mgummelt for looking into this. 

I think we should have some mechanism to show the KILLED drivers instead of making them disappear for killed command. Here finished set contains the terminated drivers, I feel even the killed drivers come under terminated drivers. Do you think, we need to show the killed drivers in a separate table?

**Comment by mgummelt** [comment]
What value is there in showing killed drivers that never ran?

**Comment by devaraj-kavali** [comment]
Killed drivers have the same value as the successfully completed drivers when we are showing in the Web UI, users would get surprised when they don't see the driver suddenly which was there in Queued/Pending for retry drivers list. I think we should treat the killed driver similar to killed application showing in the history.

**Pull request closed without merging**